javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinter.java

```diff
@@ -201,7 +205,7 @@ public void process(Node node) {
 
         // We go over tokens and find to which nodes belong. Note that we start from the most specific nodes
         // and we move up to more general nodes
-        for (JavaToken token : documentTokens) {
+        for (JavaToken token : root.getTokenRange().get()) {
```

Comment:
@ftomassetti Well, having token ranges didn't help much to reduce the code here :-)